### PR TITLE
Add Filtering to Users on Policy Details

### PR DESCRIPTION
**Pull request by adfost**
Adding filtering to Users on Policy Details 123018790-0bad9080-d384-11eb-98a5-d0fd8ce96497

### CODE REVIEW

File: portal-ui/src/screens/Console/Policies/PolicyDetails.tsx [hunk 1]
```diff
@@ -191,14 +199,22 @@ const PolicyDetails = ({
...
-          setUserList(result);
-        })
-        .catch((err) => {
-          console.log("Error in loading users");
-        });
+      if (loadingUsers) {
+        api
+          .invoke("GET", `/api/v1/policies/${policyName}/users`)
+          .then((result: any) => {
+            var resultUserArray = result.map((x: string) => {
```
Comment (dvaldivia) [inline portal-ui/src/screens/Console/Policies/PolicyDetails.tsx:206]: why not filter on an array of strings? I think it's simpler and needs less code

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: portal-ui/src/screens/Console/Policies/PolicyDetails.tsx [hunk 2]
```diff
@@ -191,14 +199,17 @@ const PolicyDetails = ({
...
-        });
+      if (loadingUsers) {
+        api
+          .invoke("GET", `/api/v1/policies/${policyName}/users`)
+          .then((result: any) => {
+            setUserList(result);
+            setLoadingUsers(false);
+          })
+          .catch((err) => {
+            console.log("Error in loading users");
```
Comment (dvaldivia) [inline portal-ui/src/screens/Console/Policies/PolicyDetails.tsx:210]: display the error when loading users.
also se `setLoadingUsers(false);` on error

File: portal-ui/src/screens/Console/Users/types.tsx [hunk 3]
```diff
@@ -15,13 +15,13 @@
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 export interface User {
-  name: string;
-  id: string;
-  email: string;
-  is_me: boolean;
-  enabled: boolean;
+  name?: string;
```
Comment (dvaldivia) [inline portal-ui/src/screens/Console/Users/types.tsx:18]: is this change still needed?

**Review by dvaldivia — CHANGES_REQUESTED** [review]

**Review by dvaldivia — APPROVED** [review]

**Review by bexsoft — APPROVED** [review]
LGTM

**Pull request merged**